cas-server-core/src/main/java/org/jasig/cas/ticket/TicketGrantingTicketImpl.java

```diff
@@ -151,6 +151,8 @@ public synchronized ServiceTicket grantServiceTicket(final String id,
         final List<Authentication> authentications = getChainedAuthentications();
         service.setPrincipal(authentications.get(authentications.size()-1).getPrincipal());
 
+        // remove this service if it already exists to only keep the latest service ticket for a service
+        this.services.values().remove(service);
```

Comment:
I was thining about the comment on the issue:

>Is this possibly an issue when running multiple web nodes behind a reverse proxy? In that case, the same web-app (service) URL can be serviced by multiple back-end servlets, so you could have multiple internal sessions for the same user-facing service.

Does that makes sense or apply?